app/post.go

```diff
@@ -179,8 +183,9 @@ func (a *App) CreatePost(post *model.Post, channel *model.Channel, triggerWebhoo
 
 	// Verify the parent/child relationships are correct
 	var parentPostList *model.PostList
-	if pchan != nil {
-		result := <-pchan
+	select {
```

Comment:
I think I prefer the older approach. Actually, is possible that after all the Sync store changes get merged into master we start to migrate the current async parts into WaitGroups instead of the current channels based approach (but we prefer this approach for now for be more systematic).

My proposal would be keep the code here as before, declare the channel without "make", and only generate the channel with make inside the same if where you launch the gorouting.

Reply:
Okay. I get it. And yeah, sync WaitGroups would be cleaner approach for handling async parts. But sometimes the channels will still be needed for async parts.